### PR TITLE
[bugfix] fix getBigDecimal() scale throwing rounding error

### PR DESCRIPTION
**Pull request by BrendanStans21**
* calculate the correct scale when calling getBigDecimal() to prevent the function throwing a rounding error

### CODE REVIEW

File: pinot-clients/pinot-jdbc-client/src/main/java/org/apache/pinot/client/PinotResultSet.java [hunk 1]
```diff
@@ -193,13 +193,19 @@ public InputStream getAsciiStream(int columnIndex)
   public BigDecimal getBigDecimal(int columnIndex, int scale)
```
Comment (walterddr) [inline pinot-clients/pinot-jdbc-client/src/main/java/org/apache/pinot/client/PinotResultSet.java:193]: i was wondering if we should just implement the non deprecated version of the API `getBigDecimal(int columnIndex/columnLabel)`?

**Review by walterddr — COMMENTED** [review]
looks good to me.

File: pinot-clients/pinot-jdbc-client/src/main/java/org/apache/pinot/client/PinotResultSet.java (hunk 1 above)
Comment (BrendanStans21) [reply pinot-clients/pinot-jdbc-client/src/main/java/org/apache/pinot/client/PinotResultSet.java:193]: It's a good question.  We want to be as minimally invasive as possible, so if your team feels that to be a necessary change, then I think it's best if they create a new issue to address it.

Comment (walterddr) [reply pinot-clients/pinot-jdbc-client/src/main/java/org/apache/pinot/client/PinotResultSet.java:193]: :+1:

**Review by walterddr — APPROVED** [review]
lgtm. tagging @xiangfu0 to take another look

**Pull request merged**